src/proxy.c

```diff
@@ -628,70 +644,51 @@ void start_bus() {
     dbus_server_setup_with_g_main (dbus_srv, NULL);
 }
 
-/*! \brief Read filter rules from JSON
- *
- * Populate the json_filters global variable with the filter rules read from
- * the JSON configuration which comes in via stdin
- *
- * \param section The section in the JSON config to use
- * \return 0      Upon success
- * \return 1      Upon failure
- */
-int parse_json_from_stdin (const char *section) {
-    size_t        i;
-    json_error_t  error;
-    json_t       *root, *config, *rule;
-    char         *full_section = calloc (sizeof (char), 30);
-    int           retval = 0;
-    int config_found = 0;
-
-    snprintf (full_section, 30, "dbus-gateway-config-%s", section);
-
-    /* Allow multiple configuration snippets, each as it's own root obj */
-    do {
-
-        /* Get root JSON object */
-        root = json_loadf (stdin, JSON_DISABLE_EOF_CHECK, &error);
-
-        if (!root) {
-            /* If no config was found this is an error, otherwise it's normal */
-            if (!config_found) {
-                g_error("error: on line %d: %s\n", error.line, error.text);
-                retval = 1;
-            }
-            break;
-        }
 
-        /* We have atleast one root object in file */
-        config_found = 1;
+void parse_full_config(const char *config_string, const char *section) {
+    json_error_t error;
+    json_t *root;
+    json_t *config;
 
-        // Get array
-        config = json_object_get (root, full_section);
+    char *full_section = calloc(sizeof(char), 30);
```

Comment:
better to use 
`calloc(30, sizeof(char));`
since the first argument indicates number of elements